### PR TITLE
Use unversioned client in scheduledjobs and set group version to batch/v2alpha1

### LINKED ISSUE #30323: Scheduled job controller can't list scheduled job
I enabled batch/v2alpha1 and can do `curl http://<master-ip>:8080/apis/batch/v2alpha1/scheduledjobs` fine. However, the scheduled job controller loop always fails at

```go
	sjl, err := jm.kubeClient.Batch().ScheduledJobs(api.NamespaceAll).List(api.ListOptions{})
```

It returns me `Error listing scheduledjobs: the server could not find the requested resource`. 

I tried switching kubeClient to unversioned.Client but it still gives me the same error. 

cc @soltysh @erictune

### PR DESCRIPTION
**Pull request by janetkuo**
Fixes #30323

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30327)

### CODE REVIEW

**Comment by soltysh** [comment]
I'll start with fixing this with #25442. The change itself LGTM.

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -128,6 +129,7 @@ func Run(s *options.CMServer) error {
 	}
 
 	kubeconfig.ContentConfig.ContentType = s.ContentType
+	kubeconfig.ContentConfig.GroupVersion = &unversioned.GroupVersion{Group: batch.GroupName, Version: "v2alpha1"}
```
Comment (soltysh) [inline cmd/kube-controller-manager/app/controllermanager.go]: On second thought, can you move this piece and initiate the kubeclient with this version only when `batch/v2alpha1` is enabled?

Comment (soltysh) [inline cmd/kube-controller-manager/app/controllermanager.go]: Iow. just before starting the controller, you do have access to kubeconfig at that point and should be able to create a new kubeclient easily.

**Comment by soltysh** [comment]
As stated before LGTM, except for the single nit which when you fix you can apply the label.

**Comment by janetkuo** [comment]
@k8s-bot test this issue: #27462

**Comment by janetkuo** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[cmd/kube-controller-manager/app/controllermanager.go, line 132 \[r1\]](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30327#-KOohDiS3iK3qYj3HBw4-r1-132:-KOqVHS9Co5dhxiuB0LT:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/d718b4565b0d6a4f7f4275096fe35943be778cb5/cmd/kube-controller-manager/app/controllermanager.go#L132)):*
<details><summary><i>Previously, soltysh (Maciej Szulik) wrote…</i></summary>
> Iow. just before starting the controller, you do have access to kubeconfig at that point and should be able to create a new kubeclient easily.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30327)*

**Comment by janetkuo** [comment]
@soltysh looks like you need to accept the change in reviewable otherwise the bot won't merge this (see the `code-review/reviewable — 3 files, 1 discussion left (soltysh)` check)

**Comment by soltysh** [comment]
LGTM

---







*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30327#-:-KOq_jME-GH2eH_VR20G:bb6ywmx)*

**Comment by soltysh** [comment]
Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/30327)*

**Comment by soltysh** [comment]
@k8s-bot test this issue: #30377

**Comment by soltysh** [comment]
@janetkuo I had one of the previous versions of this patch in #26027 which merged just now, can you update this PR with the necessary changes, currently it'll definitely conflict.

**Comment by janetkuo** [comment]
Rebased

**Comment by girishkalele** [comment]
Hopefully this commit goes in soon...the Submit Queue is blocked due to the GKE failures from #26027

**Comment by ixdy** [comment]
@girishkalele you're the buildcop, right? you can just merge it if you are satisfied it's good.

**Comment by ixdy** [comment]
(but maybe at least wait for passing PR Jenkins)

**Comment by girishkalele** [comment]
@janetkuo @ixdy We are not sure if this will fix the other e2e failures for sure.

I will revert the e2es and clear the GKE blocked tests first and then we can resubmit the reverted PR.

**Pull request merged**